sentinelsat/sentinel.py

```diff
@@ -539,16 +548,29 @@ def download(self, id, directory_path='.', checksum=True):
         if not skip_download:
             # Store the number of downloaded bytes for unit tests
             product_info['downloaded_bytes'] = self._download(
-                product_info['url'], temp_path, self.session, product_info['size'])
+                product_info['url'], temp_path, self.session, product_info['size'], product_info=product_info)
 
         # Check integrity with MD5 checksum
         if checksum is True:
-            if not self._md5_compare(temp_path, product_info['md5']):
-                remove(temp_path)
-                raise InvalidChecksumError('File corrupt: checksums do not match')
+            try:
+                if not self._md5_compare(temp_path, product_info['md5']):
+                    remove(temp_path)
+                    if product_info['title'] not in self.failed_list_title:
+                        self.failed_list_title.append(product_info['title'])
+                        self.failed_list_id.append(product_info['id'])
+                    raise InvalidChecksumError('File corrupt: checksums do not match')
+            except Exception as e:
```

Comment:
Never do this. Catchall and log is really bad practice because it obscures errors. You need to log and skip failing downloads somehow else.

Reply:
To be honest, I intentionally did it since Fin-Hub has been suffering from two major issues, one was related to some internal server issues that make the code crashes, so I was forced to wrap the methods by an error catcher otherwise I would not be able to test their services at all. The other issue with their server is that some products are missing. They are found by queries but when an HTTP request is sent to pool the product, the product can not be found, which makes python crash.
These issues forced me to use some extra error catcher in different places.

Reply:
I understand, thanks for the explanation.